MatrixSDK/Utils/Categories/FileManager+Backup.swift

```diff
@@ -0,0 +1,103 @@
+//
+// Copyright 2022 The Matrix.org Foundation C.I.C
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+public extension FileManager {
+    /**
+     Create a directory at a url, including intermediate directories, which is excluded from iCloud / manual backups, no matter where it is located.
+     
+     Note: some directories are excluded automatically if they are nested within `<AppHome>/Library/Caches/` or `<AppHome>/tmp/`
+     see [details](https://developer.apple.com/documentation/foundation/optimizing_app_data_for_icloud_backup).
+     */
+    @objc func createDirectoryExcludedFromBackup(at url: URL) throws {
+        try createDirectory(at: url, withIntermediateDirectories: true, attributes: nil)
+        try excludeFromBackup(url: url)
+    }
+    
+    /**
+     Create a directory at a path, including intermediate directories, which is excluded from iCloud / manual backups, no matter where it is located.
+     
+     Note: some directories are excluded automatically if they are nested within `<AppHome>/Library/Caches/` or `<AppHome>/tmp/`
+     see [details](https://developer.apple.com/documentation/foundation/optimizing_app_data_for_icloud_backup).
+     */
+    @objc func createDirectoryExcludedFromBackup(atPath path: String) throws {
+        try createDirectoryExcludedFromBackup(at: URL(fileURLWithPath: path))
+    }
+    
+    /**
+     Exclude a given url from iCloud / manual backups, no matter where it is located.
+     
+     Note: some directories are excluded automatically if they are nested within `<AppHome>/Library/Caches/` or `<AppHome>/tmp/`
+     see [details](https://developer.apple.com/documentation/foundation/optimizing_app_data_for_icloud_backup).
+     */
+    @objc func excludeFromBackup(url: URL) throws {
```

Comment:
I would rename this to something like `excludeURLFromBackup(_ url: URL)` to be more in line with the others.